File: openapi-generator/scala-finch/src/main/scala/org/openapitools/models/VehicleHarshEventResponseLocation.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._

/**
 * 
 * @param address Address of location where the harsh event occurred
 * @param latitude Latitude of location where the harsh event occurred
 * @param longitude Longitude of location where the harsh event occurred
 */
case class VehicleHarshEventResponseLocation(address: Option[String],
                latitude: Option[String],
                longitude: Option[String]
                )

object VehicleHarshEventResponseLocation {
    /**
     * Creates the codec for converting VehicleHarshEventResponseLocation from and to JSON.
     */
    implicit val decoder: Decoder[VehicleHarshEventResponseLocation] = deriveDecoder
    implicit val encoder: ObjectEncoder[VehicleHarshEventResponseLocation] = deriveEncoder
}

File: openapi-generator/dart-jaguar/lib/model/vehicle_harsh_event_response_location.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';

part 'vehicle_harsh_event_response_location.jser.dart';

class VehicleHarshEventResponseLocation {
   /* Address of location where the harsh event occurred */
  @Alias('address')
  final String address;
   /* Latitude of location where the harsh event occurred */
  @Alias('latitude')
  final String latitude;
   /* Longitude of location where the harsh event occurred */
  @Alias('longitude')
  final String longitude;
  

  VehicleHarshEventResponseLocation(
      

{
     this.address = null,  
     this.latitude = null,  
     this.longitude = null 
    
    }
  );

  @override
  String toString() {
    return 'VehicleHarshEventResponseLocation[address=$address, latitude=$latitude, longitude=$longitude, ]';
  }
}

@GenSerializer(nullableFields: true)
class VehicleHarshEventResponseLocationSerializer extends Serializer<VehicleHarshEventResponseLocation> with _$VehicleHarshEventResponseLocationSerializer {

}
